hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/MergeTableRegionsProcedure.java

```diff
@@ -453,13 +453,6 @@ private boolean prepareMergeRegion(final MasterProcedureEnv env) throws IOExcept
       throw new MergeRegionException("Skip merging regions " +
           RegionInfo.getShortNameToLog(regionsToMerge) + ", because we are snapshotting " + tn);
     }
-    if (!env.getMasterServices().isSplitOrMergeEnabled(MasterSwitchType.MERGE)) {
```

Comment:
You can add extra checks before scheduling the procedure to make the request fail fast, but I do not think we should remove the checks in the procedure? There could be race, that when scheduling, split and merge are enabled but when you actually execute the procedure, the flags are turned to disabled...

Reply:
Agree with Duo, others are good.

Reply:
I thought the race is not a big deal? And our admin client never have this guarantee: interrupt the schduled Split/Merge after split/merge is disabled.

Reply:
But split and merge can be scheduled by background task? And also, if you remove the checks here, then you need to add the check everywhere...

Reply:
Yes. I checked this. Now only two place can schedule split/merge task. One is the rpc request of master. Another one is from RS's report.

Reply:
But if in the future we schedule these procedures in other places, we need to add the check there...